### PR TITLE
MDS-2714 add agent now party appt

### PR DESCRIPTION
**Pull request by JustinMacaulay**
Added a new party appointment to mine party appointment type called Agent.
- available only when a NoW is imported from vfcbc.
- added section to transmogrify to import the submitter as a contact on the NoW.
- modified the contact history to show agent and link it to the NoW using the now number.
- added a new link from the NoW review for the contacts to link to the contact view page.
- Removed the option to add a new contact to a mine of type agent.

### CODE REVIEW

File: services/core-api/app/api/now_applications/transmogrify_now.py [hunk 1]
```diff
@@ -149,6 +150,57 @@ def _transmogrify_contacts(now_app, now_sub, mms_now_sub):
...
+            email=agent.email if agent.email and emailValidator.match(agent.email) else None,
+        )
+        now_party_appt_type = MinePartyAppointmentType.find_by_mine_party_appt_type_code('AGT')
+        now_party_appt = app_models.NOWPartyAppointment(
+            mine_party_appt_type_code=now_party_appt_type.mine_party_appt_type_code,
+            mine_party_appt_type=now_party_appt_type,
+            party=now_party)
+
+    if now_party_appt:
+        validPostalCode = re.compile(r"\s*([a-zA-Z]\s*\d\s*){3}$")
```
Comment (mmumby) [inline services/core-api/app/api/now_applications/transmogrify_now.py:187]: We have a different postal code regEx on the FE, should they match?
```
/^[A-Za-z]\d[A-Za-z][ -]?\d[A-Za-z]\d$/
```

Comment (JustinMacaulay) [reply services/core-api/app/api/now_applications/transmogrify_now.py:187]: that is the way both party imports work

**Review by mmumby — APPROVED** [review]
🔥

File: services/core-api/app/api/parties/response_models.py [hunk 2]
```diff
@@ -41,6 +47,21 @@
         'documents': fields.Nested(MINE_DOCUMENT)
     })
 
+NOW_APPLICATION_MODEL = api.model('NOWApplication', {
+    'now_application_guid': fields.String,
+    'now_number': fields.String,
+    'submitted_date': Date,
```
Comment (mhuseinov) [inline services/core-api/app/api/parties/response_models.py:53]: Maybe fields.Date ?

Comment (JustinMacaulay) [reply services/core-api/app/api/parties/response_models.py:53]: We use a pre made date field that allows for the proper formatting of our dates.

**Pull request merged**